celery/backends/redis.py

```diff
@@ -99,6 +101,20 @@ def on_after_fork(self):
             logger.warning(text_t(e))
         super(ResultConsumer, self).on_after_fork()
 
+    def _reconnect_pubsub(self):
+        self._pubsub = None
+        self.backend.client.connection_pool.reset()
+        # task state might have changed when the connection was down so we
+        # retrieve meta for all subscribed tasks before going into pubsub mode
+        metas = self.backend.client.mget(self.subscribed_to)
+        metas = [meta for meta in metas if meta]
+        for meta in metas:
+            self.on_state_change(self._decode_result(meta), None)
+        self._pubsub = self.backend.client.pubsub(
```

Comment:
I know this is a single statement, but the PubSub instance is also created in the `start` method. Do you think it would make sense to add a method just for that purpose?

Reply:
I don't think it's worth it, as you said it's a single statement, but if you insist I can add it.